File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/wrapper/gui/Gui.java
package rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.gui;

import rbq.wtf.lycoris.client.wrapper.MapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapMethod;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapperClass;
import rbq.wtf.lycoris.client.wrapper.wrappers.utils.ReflectUtil;
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.IWrapper;

import java.lang.reflect.Method;

@WrapperClass(mcpName = "net.minecraft.client.gui.Gui", targetMap = MapEnum.VANILLA189)

public class Gui extends IWrapper {
    @WrapMethod(mcpName = "drawModalRectWithCustomSizedTexture", targetMap = MapEnum.VANILLA189)
    public static Method drawModalRectWithCustomSizedTexture;
    @WrapMethod(mcpName = "drawRect", targetMap = MapEnum.VANILLA189)
    public static Method drawRect;

    public Gui(Object obj) {
        super(obj);
    }

    public static void drawModalRectWithCustomSizedTexture(int x, int y, float u, float v, int width, int height, float textureWidth, float textureHeight) {
        ReflectUtil.invoke(drawModalRectWithCustomSizedTexture, null, x, y, u, v, width, height, textureWidth, textureHeight);
    }

    public static void drawRect(int left, int top, int right, int bottom, int color) {
        ReflectUtil.invoke(drawRect, null, left, top, right, bottom, color);
    }
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/wrapper/gui/Gui.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.gui

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapMethod
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapperClass
import rbq.wtf.lycoris.client.wrapper.wrappers.utils.ReflectUtil
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.IWrapper
import java.lang.reflect.Method

@WrapperClass(mcpName = "net.minecraft.client.gui.Gui", targetMap = MapEnum.VANILLA189)
open class Gui(obj: Any) : IWrapper(obj) {
    companion object {
        @WrapMethod(mcpName = "drawModalRectWithCustomSizedTexture", targetMap = MapEnum.VANILLA189)
        lateinit var drawModalRectWithCustomSizedTexture: Method

        @WrapMethod(mcpName = "drawRect", targetMap = MapEnum.VANILLA189)
        lateinit var drawRect: Method

        fun drawModalRectWithCustomSizedTexture(
            x: Int,
            y: Int,
            u: Float,
            v: Float,
            width: Int,
            height: Int,
            textureWidth: Float,
            textureHeight: Float
        ) {
            ReflectUtil.invokeStatic(
                drawModalRectWithCustomSizedTexture,
                x,
                y,
                u,
                v,
                width,
                height,
                textureWidth,
                textureHeight
            )
        }

        @JvmStatic
        fun drawRect(left: Int, top: Int, right: Int, bottom: Int, color: Int) {
            ReflectUtil.invokeStatic(drawRect, left, top, right, bottom, color)
        }
    }
}